### PR TITLE
Improve Discard Changes Dialog strings

### LINKED ISSUE #15594: Improve "discard changes" dialog options
If you click Add note and type something and then close the window without saving the changes, the following message pops up.

This pop up also appears at other places (for e.g. when leaving the Deck Options page without clicking Save).

The issue with this pop up is that "OK" and "Cancel" don't seem to be the correct responses to a message that says "Close and lose current input".

What about using the following instead?

- Anki Desktop:

- Google Photos:

### PR DESCRIPTION
**Pull request by Giyutomioka-SS**
## Purpose / Description
Improved discard changes dialog option to be same as  Anki desktop 

## Fixes
* Fixes #15594 

## How Has This Been Tested?
Emulator API 31

## Screenshot's (Before & After)

image

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DiscardChangesDialog.kt [hunk 1]
```diff
@@ -17,21 +17,21 @@ package com.ichi2.anki.dialogs
...
-        ) = AlertDialog.Builder(context).show {
-            message(R.string.discard_unsaved_changes)
-            positiveButton(R.string.dialog_ok) { positiveMethod() }
-            negativeButton(R.string.dialog_cancel)
-        }
+object DiscardChangesDialog {
+    fun showDialog(
+        context: Context,
+        positiveButtonText: String,
+        negativeButtonText: String,
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DiscardChangesDialog.kt:29]: These should have defaults

**Review by david-allison — DISMISSED** [review]
Minor request for default parameters

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DiscardChangesDialog.kt [hunk 2]
```diff
@@ -17,21 +17,21 @@ package com.ichi2.anki.dialogs
...
-            message(R.string.discard_unsaved_changes)
-            positiveButton(R.string.dialog_ok) { positiveMethod() }
-            negativeButton(R.string.dialog_cancel)
-        }
+object DiscardChangesDialog {
+    fun showDialog(
+        context: Context,
+        positiveButtonText: String = "Discard",
+        negativeButtonText: String = "Keep Editing",
+        message: String,
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DiscardChangesDialog.kt:30]: and then remove the unnecessary parameters from the callers

```suggestion
        positiveButtonText: String = context.getString(R.string.discard),
        negativeButtonText: String = CollectionManager.TR.addingKeepEditing(),
        message: String = CollectionManager.TR.addingDiscardCurrentInput(),
```

File: AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateBrowserAppearanceEditor.kt [hunk 3]
```diff
@@ -98,7 +98,9 @@ class CardTemplateBrowserAppearanceEditor : AnkiActivity() {
     }
 
     private fun showDiscardChangesDialog() {
-        DiscardChangesDialog.showDialog(this, ::discardChangesAndClose)
+        DiscardChangesDialog.showDialog(this, getString(R.string.discard), CollectionManager.TR.addingKeepEditing(), CollectionManager.TR.addingDiscardCurrentInput()) {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateBrowserAppearanceEditor.kt:101]: Now you've added the defaults, you can remove the parameters which use default values

```suggestion
        DiscardChangesDialog.showDialog(this) {
```

Comment (Giyutomioka-SS) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateBrowserAppearanceEditor.kt:101]: Error is found : No value passed for parameter 'message'

Comment (Giyutomioka-SS) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateBrowserAppearanceEditor.kt:101]: Error is found : No value passed for parameter 'message'

Comment (Giyutomioka-SS) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateBrowserAppearanceEditor.kt:101]: Error found : No value passed for parameter 'message'

**Comment by Giyutomioka-SS** [comment]
Sorry for the inconvenience actually i don't have my laptop for now

**Comment by david-allison** [comment]
GitHub won't let me approve this

But I approve! Really clean

**Review by lukstbit — APPROVED** [review]
LGTM, thanks!

@david-allison I'll remove your request for changes. Also in Logcat:

 _W  Op on UI thread: anki.i18n.GeneratedTranslations$DefaultImpls.addingKeepEditing(GeneratedTranslations.kt:553)
2024-02-20 09:09:57.333  5237-5237  Backend$ch...inThreadOp com.ichi2.anki.debug                 W  Op on UI thread: anki.i18n.GeneratedTranslations$DefaultImpls.addingDiscardCurrentInput(GeneratedTranslations.kt:535)_

**Comment by dae** [comment]
That routine is safe to call from any thread and shouldn't block, so you could exclude it from the main thread check if you wanted.

**Review by mikehardy — APPROVED** [review]
LGTM - strings are clear so good to go

**Pull request merged**